lib/state.coffee

```diff
@@ -9,15 +10,19 @@ exports.findCommandBySignature = (signature) ->
 	return _.findWhere exports.commands, (command) ->
 		return command.signature.toString() is signature
 
-exports.getMatchCommand = (signature) ->
+exports.getMatchCommand = (signature, callback) ->
 
-	for command in exports.commands
+	# Omit wildcard command from the check
+	commands = _.reject exports.commands, (command) ->
+		return command.isWildcard()
 
-		# Omit wildcard command from the check
-		continue if command.isWildcard()
+	async.eachSeries commands, (command, done) ->
+		command.signature.matches signature, (result) ->
+			return done() if not result
+			return callback(null, command)
```

Comment:
Is this meant to be `callback`, this will cause async to leak memory I think, probably not an issue since this shouldn't be called much/be long running but still